### ISSUE TITLE
#returning doesn't work on a prepared dataset

### ISSUE THREAD

**Issue by celsworth** [jeremyevans/sequel#1036]
I think I've found a bug where `Dataset#returning` is ignored on a prepared dataset.

This works great:

```ruby
[16] pry(main)> dataset.where(id: 693).returning(:accessed_at).update(accessed_at: Sequel::CURRENT_TIMESTAMP)
(0.000910s) UPDATE "object_cache_entries" SET "accessed_at" = CURRENT_TIMESTAMP WHERE ("id" = 693) RETURNING "accessed_at"
=> [{:accessed_at=>2015-07-06 15:45:31 +0100}]
```

But prepare it and then call it, and the `returning` is ignored:
```ruby
[21] pry(main)> ps = dataset.where(id: :$id).returning(:accessed_at).prepare(:update, :ps_name, accessed_at: Sequel::CURRENT_TIMESTAMP)
=> <Sequel::Postgres::Dataset/PreparedStatement "UPDATE \"object_cache_entries\" SET \"accessed_at\" = CURRENT_TIMESTAMP WHERE (\"id\" = $1) RETURNING \"accessed_at\"">

[22] pry(main)> ps.call(id: 693)
(0.000276s) PREPARE ps_name AS UPDATE "object_cache_entries" SET "accessed_at" = CURRENT_TIMESTAMP WHERE ("id" = $1) RETURNING "accessed_at"
(0.001449s) EXECUTE ps_name; [693]
=> 1
```

Would be great if this could be fixed. If it can't, the documentation could do with updating to point this out as there is no mention anywhere that `returning` won't work with prepared statements.

**Comment by celsworth** [comment]
In IRC a suggested solution was `with_sql`, but this blows up, the prepared statement appears to be losing the `:$id` placeholder:

```ruby
[8] pry(main)> ps = Model::ObjectCacheEntry.dataset.where(id: :$id).returning(:accessed_at, :updated_at).with_sql(:update_sql, accessed_at: Sequel::CURRENT_TIMESTAMP).prepare(:update, :foobar)
=> <Sequel::Postgres::Dataset/PreparedStatement "UPDATE \"object_cache_entries\" SET \"accessed_at\" = CURRENT_TIMESTAMP WHERE (\"id\" = \"$id\") RETURNING \"accessed_at\", \"updated_at\"">
 
[9] pry(main)> ps.call(id: 693)
PG::UndefinedColumn: ERROR:  column "$id" does not exist
LINE 1: ...T "accessed_at" = CURRENT_TIMESTAMP WHERE ("id" = "$id") RET...
                                                             ^: PREPARE foobar AS UPDATE "object_cache_entries" SET "accessed_at" = CURRENT_TIMESTAMP WHERE ("id" = "$id") RETURNING "accessed_at", "updated_at"
Sequel::DatabaseError: PG::UndefinedColumn: ERROR:  column "$id" does not exist
LINE 1: ...T "accessed_at" = CURRENT_TIMESTAMP WHERE ("id" = "$id") RET...
                                                             ^
from /home/chris/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/sequel-4.24.0/lib/sequel/adapters/postgres.rb:582:in `prepare'
```

**Comment by jeremyevans** [comment]
I think it's possible to fix this.  I'll try to work on it soon.

**Comment by jeremyevans** [comment]
Fixed by b6accd8aa319e73886addba89c96fb3f359c7f47
